src/components/githubDomainChecker.tsx

```diff
@@ -0,0 +1,96 @@
+'use client';
+
+import {useState} from 'react';
+
+const MAX_COMPONENTS_ON_PAGE = 100;
+
+export function GitHubDomainChecker() {
+  const [domain, setDomain] = useState('');
+  const [isValidDomain, setIsValidDomain] = useState(false);
+
+  const isGitHubCom =
+    domain.toLowerCase().trim() === 'github.com' ||
+    domain.toLowerCase().trim() === 'https://github.com';
+  const hasInput = domain.trim().length > 0;
+
+  // Validate domain format
+  const validateDomain = (inputDomain: string) => {
+    const trimmedDomain = inputDomain.trim();
+    if (!trimmedDomain) {
+      setIsValidDomain(false);
+      return;
+    }
+
+    // Check if it's github.com (valid)
+    if (
+      trimmedDomain.toLowerCase() === 'github.com' ||
+      trimmedDomain.toLowerCase() === 'https://github.com'
+    ) {
+      setIsValidDomain(true);
+      return;
+    }
+
+    // For enterprise, check if it's a valid URL or domain format
+    const urlPattern = /^(https?:\/\/)?([\w\-\.]+\.[\w]{2,})(\/.*)?$/;
+    const isValidUrl = urlPattern.test(trimmedDomain);
+    setIsValidDomain(isValidUrl);
+  };
+
+  const handleDomainChange = ev => {
+    const newDomain = ev.target.value;
+    setDomain(newDomain);
+    validateDomain(newDomain);
+  };
+
+  const inputClassName =
+    'form-input w-full rounded-md border-[1.5px] focus:ring-2 focus:ring-accent-purple/20 border-gray-200';
+
+  // This is to avoid in case multiple instances of this component are used on the page
+  const randomCounter = Math.round(Math.random() * MAX_COMPONENTS_ON_PAGE);
```

Comment:
This is an interesting pattern! could we avoid it by passing in a `key` or `id` prop when rendering, or is that like a limitation of MDX?